pwiz_tools/Skyline/Model/Lib/BlibData/BlibDb.cs

```diff
@@ -136,6 +136,90 @@ public int GetSpectraCount()
             }
         }
 
+        private class ProteinTablesBuilder
+        {
+            private ISession _session;
+            private Dictionary<string, int> _namedProteinIds;
+            private Dictionary<int, int> _peptideIdProteinId;
+
+            public ProteinTablesBuilder(ISession session)
+            {
+                _session = session;
+                _namedProteinIds = new Dictionary<string, int>();
+                _peptideIdProteinId = new Dictionary<int, int>();
+            }
+
+            public void Add(DbRefSpectra refSpectra, string proteinName)
+            {
+                if (!string.IsNullOrEmpty(proteinName))
+                {
+                    if (!_namedProteinIds.TryGetValue(proteinName, out var proteinTableId))
+                    {
+                        proteinTableId = _namedProteinIds.Count + 1;
+                        _namedProteinIds.Add(proteinName, proteinTableId);
+                    }
+                    _peptideIdProteinId.Add((int)(refSpectra.Id ?? 0), proteinTableId);
+                }
+            }
+
+            public void Write()
+            {
+                // Output the protein - peptide relationships
+                using (var cmd = _session.Connection.CreateCommand())
+                {
+                    cmd.CommandText = @"DROP TABLE IF EXISTS Proteins";

```

Comment:
It does not seem right to drop a table like this. I imagine it does not matter because this only ever gets called on blank databases. But, it seems like the right thing to do would be if the table already exists, just assume it has the right schema, and append your new rows to it.

Reply:
I suppose you're right. I'll have to inspect for available IDs.